File: 1758-minimum-changes-to-make-alternating-binary-string/1758-minimum-changes-to-make-alternating-binary-string.cpp
class Solution {
public:
    int minOperations(string s) {
        int ans = 0;

        // Check for alternating pattern
        for (int i = 0; i < s.size(); i++) {
            if (s[i] != (i % 2 == 0 ? '0' : '1')) {
                ans++;
            }
        }

        // Return the minimum of the two counts
        return std::min(ans, static_cast<int>(s.size()) - ans);
    }
};

File: 1758-minimum-changes-to-make-alternating-binary-string/1758-minimum-changes-to-make-alternating-binary-string.py
class Solution(object):
    def minOperations(self, s):
        """
        :type s: str
        :rtype: int
        """
        count = 0
        
        for i in range(len(s)):
            if i % 2: #odd
                count += 1 if s[i] == "0" else 0
            else: #even
                count += 1 if s[i] == "1" else 0
        return min(count, len(s) - count)
        